### ISSUE TITLE
prefer-screen-queries doesn't work with `within`

### ISSUE THREAD

**Issue by benmonro** [testing-library/eslint-plugin-testing-library#241]
ran into a false negative that I thought i'd share.  consider the following code: 

```jsx
import { screen, within } from "@testing-library/react";

const container = within(screen.getByRole("foo"));

expect(container.getByText("bar"));
expect(container.getByText("baz"));
```

these are being reported as invalid, but they are valid. can this be supported in the rule?

**Comment by gndelia** [comment]
According to the [docs](https://github.com/testing-library/eslint-plugin-testing-library/blob/master/docs/rules/prefer-screen-queries.md) these are valid and supported 🤔 

I don't see an exact test matching this scenario, I will try to take a look in the incoming days

**Comment by gndelia** [comment]
Hey, @benmonro  I created this ☝️  PR with your scenario and, unless I'm missing something, it's passing. Maybe you need to upgrade to a newer version? Which version are you using of the plugin?

In addition to that, could you verify if the code is exactly that one, or if the rule that fails is this one or another? Thanks!

**Comment by benmonro** [comment]
yeah doesn't seem to be an issue anymore, if it pops up again i'll reopen
